src/App.jsx

```diff
@@ -120,23 +127,29 @@ function KanbanBoard() {
 
       const updatedColumns = newOrder.reduce((acc, col, idx) => ({
         ...acc,
-        [col.id]: {
+        [String(col.id)]: {
           ...col,
           order: idx
         }
       }), {});
 
+      console.log('Atualizando ordem:', updatedColumns);
       await updateColumnsOrder(updatedColumns);
       return;
     }
 
-    await moveTask(
-      source.droppableId,
-      destination.droppableId,
-      result.draggableId,
-      source.index,
-      destination.index
-    );
+    // Movendo tasks
+    try {
+      await moveTask(
+        String(source.droppableId),
+        String(destination.droppableId),
+        String(result.draggableId),
+        source.index,
+        destination.index
+      );
+    } catch (error) {
+      console.error('Erro ao mover task:', error);
+    }
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```javascript
try {
  await moveTask(
    String(source.droppableId),
    String(destination.droppableId),
    String(result.draggableId),
    source.index,
    destination.index
  );
} catch (error) {
  console.error('Error moving task:', error);
  // Provide user feedback
  if (error.code === 'PERMISSION_DENIED') {
    // Handle permission error
  } else if (error.code === 'INVALID_OPERATION') {
    // Handle invalid operation
  }
}
```
The error handling in the drag-and-drop operation could be improved with more specific error types and user feedback.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;